beacon-chain/sync/rpc_send_request.go

```diff
@@ -198,6 +205,50 @@ func SendBlobSidecarByRoot(
 
 type blobResponseValidation func(blocks.ROBlob) error
 
+func chainBlobValidations(vf ...blobResponseValidation) blobResponseValidation {
+	return func(blob blocks.ROBlob) error {
+		for i := range vf {
+			if err := vf[i](blob); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+type seqBlobValid struct {
+	prev *blocks.ROBlob
+}
+
+func (sbv *seqBlobValid) nextValid(blob blocks.ROBlob) error {
+	if sbv.prev == nil {
+		sbv.prev = &blob
+		return nil
+	}
+	if sbv.prev.BlockRoot() == blob.BlockRoot() {
```

Comment:
Another thing. If the block root is the same, it won't check the blob's slot and parentBlockRoot. It might be useful to have another function which ensures blockRoot, parentBlockRoot, and slot are equal.